### PR TITLE
Fix crash in roundrobin assignor for asymmetrical subscriptions (#2121)

### PR DESCRIPTION
**Pull request by edenhill**
This also adds declarative unit tests of the assignors.

### CODE REVIEW

File: src/rdkafka_assignor.c [hunk 1]
```diff
@@ -558,3 +559,403 @@ int rd_kafka_assignors_init (rd_kafka_t *rk, char *errstr, size_t errstr_size) {
...
+                        char errstr[256];
+
+                        RD_UT_SAY("Test case %s: %s assignor",
+                                  tests[i].name,
+                                  tests[i].expect[ie].protocol_name);
+
+                        /* Run assignor */
+                        err = rd_kafka_assignor_run(
+                                rk->rk_cgrp,
+                                tests[i].expect[ie].protocol_name,
```
Comment (mhowlett) [inline src/rdkafka_assignor.c:866]: heads up that i've changed this to be a pointer to the assignor in the change for stateful assignors. also, `rk->rk_cgrp` maintains this state so it's kind of superfluous to specify it separately. anyway fothcoming change.

File: src/rdkafka_assignor.c [hunk 2]
```diff
@@ -208,7 +209,7 @@ int rd_kafka_assignor_topic_cmp (const void *_a, const void *_b) {
         const rd_kafka_assignor_topic_t *b =
                 *(const rd_kafka_assignor_topic_t * const *)_b;
 
-        return !strcmp(a->metadata->topic, b->metadata->topic);
+        return strcmp(a->metadata->topic, b->metadata->topic);
```
Comment (edenhill) [inline src/rdkafka_assignor.c:212]: Found this issue (which existed prior to this PR).

Comment (mhowlett) [reply src/rdkafka_assignor.c:212]: LGTM. we need better testing!

Comment (edenhill) [reply src/rdkafka_assignor.c:212]: It was found by the tests added in this PR ;)

**Pull request merged**